File: gitnexus/test/fixtures/lang-resolution/python-fastapi-handler-service/services/order_service.py
from typing import Optional


class OrderService:
    def __init__(self):
        self._orders: dict[int, dict] = {}

    def get_order(self, order_id: int) -> Optional[dict]:
        return self._orders.get(order_id)

    def delete_order(self, order_id: int) -> None:
        self._orders.pop(order_id, None)


def get_order_service() -> OrderService:
    return OrderService()

File: gitnexus/test/fixtures/lang-resolution/go-implements-cross-file/services/order_service.go
// WI-H85 fixture (issue #85): OrderService struct that satisfies
// IOrderService from interfaces/service_interface.go. Exercises
// multiple cross-file matches in a single fixture.
package services

type OrderService struct {
	orders map[int]map[string]interface{}
}

func (s *OrderService) GetOrder(id int) *map[string]interface{} {
	o, ok := s.orders[id]
	if !ok {
		return nil
	}
	return &o
}

func (s *OrderService) DeleteOrder(id int) bool {
	if _, ok := s.orders[id]; !ok {
		return false
	}
	delete(s.orders, id)
	return true
}
